src/digital.rs

```diff
@@ -123,3 +123,112 @@ pub trait InputPin {
     /// Is the input pin low?
     fn try_is_low(&self) -> Result<bool, Self::Error>;
 }
+
+/// Dummy GPIO pin
+///
+/// These structures are useful when using optional pins, for example
+/// when using some SPI devices.
+pub mod dummy {
+    use super::{toggleable, InputPin, OutputPin, StatefulOutputPin};
+    use core::{convert::Infallible, marker::PhantomData};
+
+    /// Dummy input pin level
+    pub mod level {
+        /// Dummy input pin will always behave as being high.
+        pub struct High;
+        /// Dummy input pin will always behave as being low.
+        pub struct Low;
+    }
+
+    /// Dummy (No-op) pin
+    ///
+    /// This will discard any value set to it and when checked always behave
+    /// according to the value provided at construction time
+    pub struct DummyPin<L = level::Low> {
+        _l: PhantomData<L>,
+    }
+
+    impl DummyPin<level::Low> {
+        /// Create new instance
+        ///
+        /// When checked it will always behave as being low
+        pub fn new() -> Self {
+            DummyPin { _l: PhantomData }
+        }
+
+        /// Create new instance
+        ///
+        /// When checked it will always behave as being low
+        pub fn new_low() -> Self {
+            DummyPin { _l: PhantomData }
+        }
```

Comment:
Why two identical constructors `new` and `new_low`?

Reply:
I thought it would be appropriate to have `new_high()` and `new_low()` if the user is interested in using it as an `InputPin` since this is more explicit than the default `new()`.
Nevertheless, somebody that just wants an `OutputPin` can rather expect a `new()` method.
I should define one by using the other, though.

Reply:
`new_low()` and `new_high()` seem perfectly fine to me. I was confused by `new()`. 😅

Reply:
ok, sure.